### PR TITLE
Add support for oracle functions and object storage

### PR DESCRIPTION
**Pull request by abourramouss**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

This pull request adds support for the oracle backend using lithops.

### CODE REVIEW

**Review by JosepSampe — CHANGES_REQUESTED** [review]
Thanks for the contribution @abourramouss. 

Please, add only files related to oracle and remove all the modifications in all the rest

File: .github/workflows/python-app.yml
Comment (JosepSampe) [inline .github/workflows/python-app.yml:1]: Please remove this workflow

File: Dockerfile
Comment (JosepSampe) [inline Dockerfile:1]: Why is this file located in the root of lithops? Please remove this file

File: examples/call_async.py
Comment (JosepSampe) [inline examples/call_async.py:1]: Please remove the changes on this file from this PR

File: examples/call_async_class.py
Comment (JosepSampe) [inline examples/call_async_class.py:1]: Please remove the changes on this file from this PR

File: examples/call_async_java.py
Comment (JosepSampe) [inline examples/call_async_java.py:1]: Please remove the changes on this file from this PR

File: examples/ibmcos_arg.py
Comment (JosepSampe) [inline examples/ibmcos_arg.py:1]: Please remove the changes on this file from this PR

File: examples/call_async_cython.py
Comment (JosepSampe) [inline examples/call_async_cython.py:1]: Please remove the changes on this file from this PR

File: examples/function.cpython-39-x86_64-linux-gnu.so
Comment (JosepSampe) [inline examples/function.cpython-39-x86_64-linux-gnu.so:1]: Please remove the changes on this file from this PR

File: examples/multicloud.py
Comment (JosepSampe) [inline examples/multicloud.py:1]: Please remove the changes on this file from this PR

File: examples/rabbitmq_arg.py
Comment (JosepSampe) [inline examples/rabbitmq_arg.py:1]: Please remove the changes on this file from this PR

File: examples/wait.py
Comment (JosepSampe) [inline examples/wait.py:1]: Please remove the changes on this file from this PR

File: lithops/serverless/backends/aliyun_fc/aliyun_fc.py
Comment (JosepSampe) [inline lithops/serverless/backends/aliyun_fc/aliyun_fc.py:1]: Please remove the changes on this file from this PR

File: lithops/serverless/backends/aws_lambda/aws_lambda.py
Comment (JosepSampe) [inline lithops/serverless/backends/aws_lambda/aws_lambda.py:1]: Please remove the changes on this file from this PR

File: lithops/serverless/backends/azure_functions/entry_point.py
Comment (JosepSampe) [inline lithops/serverless/backends/azure_functions/entry_point.py:1]: Please remove the changes on this file from this PR

File: runtime/aws_lambda/Dockerfile
Comment (JosepSampe) [inline runtime/aws_lambda/Dockerfile:1]: Please remove the changes on this file from this PR

File: setup.py [hunk 1]
```diff
@@ -23,7 +23,8 @@
     'ibm-vpc',
     'cloudpickle',
     'tblib',
-    'ps-mem'
+    'ps-mem',
+    'oci'
```
Comment (JosepSampe) [inline setup.py:27]: Please remove this line

**Review by JosepSampe — CHANGES_REQUESTED** [review]
Thanks for the contribution @abourramouss.

Please, add only files related to oracle and remove all the modifications in all the rest

**Review by JosepSampe — CHANGES_REQUESTED** [review]
Can you add the copyright header to all the .py files (except \__init__.py files)?

File: lithops/storage/backends/oracle_oss/config.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#
+# (C) Copyright Cloudlab URV 2020
```
Comment (JosepSampe) [inline lithops/storage/backends/oracle_oss/config.py:2]: Should be 2023?

File: docs/source/compute_config/oracle_functions.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    key_file : <KEY_FILE>
+    fingerprint : <FINGERPRINT>
+    tenancy : <TENANCY>
+    region : <REGION>
+    compartment_id: <COMPARTMENT_ID>
+    namespace_name : <NAMESPACE_NAME>
+oracle_f:
+    max_workers: <MAX_WORKERS>
+    runtime: <RUNTIME>
+    runtime_memory: <RUNTIME_MEMORY>
```
Comment (JosepSampe) [inline docs/source/compute_config/oracle_functions.md:90]: You can remove max_workers, runtime and runtime_memory from here if they are not mandatory

File: docs/source/storage_config/oracle_oss.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+1. Navigate to the [Oracle Cloud Console](https://console.oraclecloud.com/) and create a new API signing keys (If you don't have one)
+
+2. Edit your Lithops config and add the following keys:
+
+```yaml
+lithops:
+    storage : oracle_oss
+
+oracle:
+    storage_bucket : <STORAGE_BUCKET>
```
Comment (JosepSampe) [inline docs/source/storage_config/oracle_oss.md:20]: is `storege_bucket` needed here?

File: lithops/serverless/backends/oracle_f/oracle_f.py [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+            raise Exception("Application %s not found.", app_id)
+
+        # Get the list of functions within the found application
+        functions = self.cf_client.list_functions(app_id).data
+
+        # Search for the function and return its OCID
+        for function in functions:
+            if function.display_name == runtime_name:
+                return function.id
+        raise Exception("Function %s not found.", runtime_name)
```
Comment (JosepSampe) [inline lithops/serverless/backends/oracle_f/oracle_f.py:319]: Add a blank line at the end of the file

File: lithops/serverless/backends/oracle_f/config.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    for key in DEFAULT_CONFIG_KEYS:
+        if key not in config_data['oracle_f']:
+            config_data['oracle_f'][key] = DEFAULT_CONFIG_KEYS[key]
+
+    if 'vcn' not in config_data['oracle_f'] or 'subnet_ids' not in config_data['oracle_f']['vcn']:
+        raise Exception("'vcn' and 'subnet_ids' are mandatory in the 'oracle_f' section of the configuration")
+    else:
+        config_data['oracle_f']['subnet_ids'] = config_data['oracle_f']['vcn']['subnet_ids']
+
+    config_data['oracle_f'].update(config_data['oracle'])
```
Comment (JosepSampe) [inline lithops/serverless/backends/oracle_f/config.py:50]: Add a blank line at the end of the file

File: lithops/serverless/backends/oracle_f/config.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+CONNECTION_POOL_SIZE = 300
+
+APPLICATION_NAME = 'lithops'
+BUILD_DIR = os.path.join(TEMP_DIR, 'OracleRuntimeBuild')
+
+AVAILABLE_PY_RUNTIMES = {
+    '3.8': 'python3',
+}
```
Comment (JosepSampe) [inline lithops/serverless/backends/oracle_f/config.py:35]: Is Oracle functions available only for python3.8? I can see you added a Dockefile, so: does this mean Oracle Functions can work with any Python version?

[the remark this answers is not in the document]
Comment (abourramouss) [reply lithops/serverless/backends/oracle_f/config.py:35]: It actually depends on the oci-sdk, it must run on any version the oci-sdk supports.
The oci-sdk supports Python 3.6, 3.7, 3.8 and 3.9.

**Pull request merged**